File: GenesisAndroid/src/main/java/com/emerchantpay/gateway/genesisandroid/api/internal/TransactionGateway.java
package com.emerchantpay.gateway.genesisandroid.api.internal;

import com.emerchantpay.gateway.genesisandroid.api.models.Transaction;

import com.emerchantpay.gateway.genesisandroid.api.util.Configuration;
import com.emerchantpay.gateway.genesisandroid.api.util.NodeWrapper;

public class TransactionGateway {

    Configuration configuration;
    NodeWrapper response;


    public TransactionGateway(Configuration configuration, NodeWrapper response) {
        super();

        this.configuration = configuration;
        this.response = response;
    }

    public TransactionResult<Transaction> getRequest() {
        return new TransactionResult(response, Transaction.class);
    }
}

File: GenesisAndroid/src/main/java/com/emerchantpay/gateway/genesisandroid/api/internal/TransactionGateway.kt
package com.emerchantpay.gateway.genesisandroid.api.internal

import com.emerchantpay.gateway.genesisandroid.api.models.Transaction

import com.emerchantpay.gateway.genesisandroid.api.util.Configuration
import com.emerchantpay.gateway.genesisandroid.api.util.NodeWrapper

open class TransactionGateway(internal var configuration: Configuration?, internal var response: NodeWrapper) {

    val request: TransactionResult<Transaction>
        get() = TransactionResult(response, Transaction::class.java)
}
